src/Jab/DiagnosticDescriptors.cs

```diff
@@ -50,4 +50,12 @@ internal static class DiagnosticDescriptors
         "The factory member has to be a method or have a delegate type",
         "The factory member '{0}' has to be a method of have a delegate type, for service '{1}'", "Usage", DiagnosticSeverity.Error, true);
 
+    public static readonly DiagnosticDescriptor NullableServiceNotRegistered = new("JAB0013",
+        "Not registered nullable dependency without a default value",
+        "'{0}' parameter to construct '{1}' will always be null when constructing using a service provider. Add a default value to make the service reference optional", "Usage", DiagnosticSeverity.Warning, true);
```

Comment:
The error message is not entirely correct. The project will never build when a service dependency is not satisfied. ServiceRequiredToConstructNotRegistered error will fire.

Reply:
Actually I've done in a way that it does build

Reply:
Hm, I missed that. It shouldn't build, required parameter == required dependencies is one of the core invariants of Jab

Reply:
The problem is third-party libraries as from Microsoft where you can't control the constructor. As the `LoggerFactory` and `ConsoleLoggerProvider` that defines nullable parameters as optional dependency.

Reply:
It is important to note that the change is only when parameter is nullable (`?`), otherwise the build will fail.

Reply:
Another option is defining a MSBuild property to enable this behavior.

```xml
<PropertyGroup>
  <JabUseNullableParametersAsOptional>true</JabUseNullableParametersAsOptional>
</PropertyGroup>
```

Reply:
Yeah, but there are other constructors, and DI container must pick the longest satisfiable one. In both of those cases, there are shorter overloads without nullable parameters.

Reply:
Yes, you're right. I think that some misbehavior was occurred. I undone the change to handle nullable as optional.